synapseclient/client.py

```diff
@@ -2098,15 +2098,21 @@ def getPermissions(self, entity, principalId=None):
                   or an empty array
```

Comment:
There are 3 functions that are assuming some typing here - `str`, `Evaluation` are covered early on in the call chain, then we get to this method which is adding in `Entity` and a `collections.abc.Mapping`:

```
ENTITY_TYPES = [
    "org.sagebionetworks.repo.model.FileEntity",
    "org.sagebionetworks.repo.model.Folder",
    "org.sagebionetworks.repo.model.Link",
    "org.sagebionetworks.repo.model.Project",
    "org.sagebionetworks.repo.model.table.TableEntity",
    "org.sagebionetworks.repo.model.table.Dataset",
]


def is_synapse_entity(entity):
    if isinstance(entity, Entity):
        return True
    if isinstance(entity, collections.abc.Mapping):
        return entity.get("concreteType", None) in ENTITY_TYPES
    return False
```

That tells me that we should be expecting a typing of:
`entity: Union[Entity, Evaluation, str, collections.abc.Mapping]`

Reply:
Yes, technically since the `Entity` type extends the `dict` type, you can technically pass in a `dict` or `Mapping` of any sort and the function would work